tools/mo/openvino/tools/mo/main.py

```diff
@@ -321,6 +323,47 @@ def load_extensions(argv: argparse.Namespace, is_tf: bool, is_caffe: bool, is_mx
 def check_fallback(argv : argparse.Namespace):
     fallback_reasons = {}
 
+    def legacy_extensions_used(argv):
+        any_extension_used = hasattr(argv, 'extensions') and argv.extensions is not None and len(argv.extensions) > 0 \
+            and argv.extensions != import_extensions.default_path() # extensions arg has default value
+        if not any_extension_used:
+            return False
+        path = Path(argv.extensions)
+        new_ext_used = path.is_file() and (path.suffix == '.so' or path.suffix == '.dll')
+        if new_ext_used:
+            return False
+        return True
+
+    def is_new_json_config(json_file_path: Path):
+        with open(json_file_path) as stream:
+            config_content = json.load(stream)
+            if len(config_content) == 0: # empty case
+                return False
+            if isinstance(config_content, dict): # single transformation
+                return 'library' in config_content.keys()
+            # many transformations in single file
+            library_counter = 0
+            for transform in config_content:
+                if any(key == 'library' for key in transform.keys()):
+                    library_counter+=1
+            if len(config_content) == library_counter: # all transformations has 'library' attribute
+                return True
+            elif library_counter == 0: # all transformations are legacy type
+                return False
+            else:
+                raise Error('Mixed types of transformations configurations were used')
+
+    def legacy_transformations_config_used(argv):
```

Comment:
Also, we need to check that an user does not pass JSON with shared library to legacy MO. Propose to check it in `main.py` in `arguments_post_parsing`. If one of variable `is_tf`, etc is True and JSON with shared library is specified raise an Error. We need to do the same stuff for `--extensions`

Reply:
> Also, we need to check that an user does not pass JSON with shared library to legacy MO.
Based on current arrangements, the only way to use a shared library in JSON on the new path is via 'library' field. The field is applicable (@itikhono could you comment?) only for the new frontend so additional checking seems to be not needed.
> We need to do the same stuff for --extensions
I am waiting here to decision about indicating the directory or shared lib path itsefl. In the second case, an additional check seems to be not needed.

Reply:
> I am waiting here to decision about indicating the directory or shared lib path itsefl. In the second case, an additional check seems to be not needed.
@mbencer, please create email thread to discuss it